crypto/hashes.go

```diff
@@ -65,6 +66,29 @@ type HashFactory struct {
 
 var errUnknownHash = errors.New("unknown hash type")
 
+func (h HashType) String() string {
+	switch h {
+	case Sha512_256:
+		return "sha512_256"
+	case Sumhash:
+		return "sumhash"
+	default:
+		return ""
+	}
+}
+
+// InitHashType decodes a string into the HashType enum
+func InitHashType(s string) (HashType, error) {
```

Comment:
`UnmarshalHashType` would be a more consistent name for this.

The `Status` type from `data/basics/userBalance.go` already has `String` and `UnmarshalStatus` functions.